tests/components/vicare/test_config_flow.py

```diff
@@ -45,12 +45,15 @@ async def test_form(hass: HomeAssistant) -> None:
     assert len(mock_setup_entry.mock_calls) == 1
 
 
-async def test_invalid_login(hass: HomeAssistant) -> None:
+async def test_invalid_login(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+) -> None:
```

Comment:
You should not have a test finish on "FORM". Please remove this test and combine it with the above:
- init
- configure (bad)
- configure (good)
- success

Reply:
vesync config flow tests also have this sort of test test: https://github.com/home-assistant/core/blob/8475acebd1ab370acd66aa6c7bff383811470ce8/tests/components/vesync/test_config_flow.py#L25 

is there a good example on how to structure config flow tests with a cloud API?

Reply:
You can look at `nextcloud`, which recently added config flow.
You will see that all tests start with async_init and finish with `CREATE_ENTRY`/`ABORT_ENTRY`. They neven finish with just `FORM`.

https://github.com/home-assistant/core/blob/565f311f5c7b7d20adee8e0445ae5ff48eb6cd4a/tests/components/nextcloud/test_config_flow.py#L31-L33

and

https://github.com/home-assistant/core/blob/565f311f5c7b7d20adee8e0445ae5ff48eb6cd4a/tests/components/nextcloud/test_config_flow.py#L195-L197